packages/flutter_tools/lib/src/flx.dart

```diff
@@ -115,7 +115,13 @@ Map<_Asset, List<_Asset>> _parseAssets(
 
       List<FileSystemEntity> files = assetDir.listSync(recursive: true);
 
+      final String buildDirPath = path.absolute(path.join(assetBase, 'build'));
```

Comment:
The directory search for variants should be rooted at the directory where the main asset is located. Can you give an example where you believe this search would be too permissive? Forcing the variants to follow a too-specific convention now risks brittleness when we start adding variants along more dimensions than just device pixel ratio.

I'm fine adding a convention for where all assets live, and preventing other artifacts from living there too. Most systems have that. But I don't know if that would address your concern.

I'd prefer not to have to include all asset variants by name; that removes a considerable part of the convenience.